yearn/contracts/SaddleStrategy.sol

```diff
@@ -0,0 +1,395 @@
+
+    function wrappedToken() external view returns (address);
+}
+
+/// @notice Interface for the Uniswap v2 router.
+/// @dev This is an interface with just a few function signatures of the
+///      router contract. For more info and function description please see:
+///      https://uniswap.org/docs/v2/smart-contracts/router02.
+interface IUniswapV2Router {
+    function swapExactTokensForTokens(
+        uint256,
+        uint256,
+        address[] calldata,
+        address,
+        uint256
+    ) external;
+}
+
+/// @title SaddleStrategy
+/// @notice This strategy is meant to be used with the Saddle tBTC v2 pool vault.
+///         The vault's underlying token (a.k.a. want token) should be the LP
+///         token of the tBTC v2 Saddle pool. This strategy borrows the vault's
+///         underlying token up to the debt limit configured for this strategy
+///         in the vault. In order to make the profit, the strategy deposits
+///         the borrowed tokens into the tBTC v2 Saddle reward pool. Depositing
+///         tokens in the reward pool generates KEEP token rewards.
+///         The financial outcome is settled upon a call of the `harvest` method
+///         (BaseStrategy.sol). Once that call is made, the strategy gets the
+///         KEEP token rewards from Saddle reward pool. These reward tokens are
+///         then used to buy wBTC via a decentralized exchange. At the end, the
+///         strategy takes acquired wBTC and deposits them to the Saddle tBTC v2
+///         pool. This way it obtains new LP tokens the vault is interested in,
+///         and makes the profit in result. At this stage, the strategy may
+///         repay some debt back to the vault, if needed. The entire cycle
+///         repeats for the strategy lifetime, so all gains are constantly
+///         reinvested. Worth to flag that the current implementation uses wBTC
+///         as the intermediary token because of its liquidity and ubiquity in
+///         BTC-based Curve pools.
+/// @dev Implementation is based on:
+///      - General Yearn strategy template
+///        https://github.com/yearn/brownie-strategy-mix
+contract SaddleStrategy is BaseStrategy {
+    using SafeERC20 for IERC20;
+    using Address for address;
+    using SafeMath for uint256;
+
+    // Address of the KEEP token contract.
+    address public constant keep = 0x85Eee30c52B0b379b046Fb0F85F4f3Dc3009aFEC;
+    // Address of the WETH token contract.
+    address public constant weth = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2;
+    // Address of the WBTC token contract.
+    address public constant wbtc = 0x2260FAC5E5542a773Aa44fBCfeDf7C193bc2C599;
+    // Address of the Uniswap V2 router contract.
+    address public constant uniswap =
+        0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D;
+    // Address of the Saddle Swap contract, taken from
+    // https://docs.saddle.finance/contracts
+    address public constant swap = 0x4f6A43Ad7cba042606dECaCA730d4CE0A57ac62e;
```

Comment:
What do you think about naming it `tbtcSaddlePoolSwap` and make it a variable set using the constructor? Regarding the name, `swap` seems too broad and it took me a while to figure out what it refers to. Regarding making it a variable, we don't know yet how tBTC v2 will be added to Saddle. It may be added to the existing BTC pool but it may require a new pool (thus a new swap contract) as well. Making it settable via the constructor is safer.

Reply:
Done in https://github.com/keep-network/tbtc-v2/pull/13/commits/227ed9911ad590b61480bb840375083958d2233b.